File: fixtures/scale/ruby/large/lib/check_affine.rb
require_relative 'affine_encrypt'
require_relative 'affine_decrypt'
require_relative 'corpus_affine'

def check_affine
  corpus_affine.each do |text|
    sealed = affine_encrypt(text, 11)
    opened = affine_decrypt(sealed, 11)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_affine.py
from cipherlab.ciphers.affine import affine_encrypt, affine_decrypt
from cipherlab.specs.affine_spec import affine_spec
from cipherlab.corpus.corpus_affine import corpus_affine


def check_affine():
    spec = affine_spec()
    for text in corpus_affine():
        sealed = affine_encrypt(text, spec["key"])
        opened = affine_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "affine"
